test/simple_mc_integration.cpp

```diff
@@ -0,0 +1,271 @@
+// VolEsti (volume computation and sampling library)
+
+// Copyright (c) 2012-2021 Vissarion Fisikopoulos
+// Copyright (c) 2018-2021 Apostolos Chalkis
+
+// Contributed and/or modified by Suraj Choubey, as part of Google Summer of Code 2021 program.
+
+// Licensed under GNU LGPL.3, see LICENCE file
+
+#include "doctest.h"
+#include "simple_MC_integration.hpp"
+#include "Eigen/Eigen"
+#include <vector>
+#include "cartesian_geom/cartesian_kernel.h"
+#include "convex_bodies/hpolytope.h"
+#include "generators/known_polytope_generators.h"
+#include "ode_solvers/oracle_functors.hpp"
+#include "random_walks/random_walks.hpp"
+#include <iostream>
+#include <fstream>
+#include "misc.h"
+
+typedef double NT;
+typedef Cartesian<NT> Kernel;
+typedef typename Kernel::Point Point;
+typedef std::vector<Point> Points;
+typedef HPolytope<Point> HPOLYTOPE;
+typedef VPolytope<Point> VPOLYTOPE;
+typedef boost::mt19937 RNGType;
+typedef BoostRandomNumberGenerator<RNGType, NT> RandomNumberGenerator;
+
+NT exp_normsq(Point X) {
+	return exp(-X.squared_length()) ;
+}
+
+NT simple_polynomial_1D(Point X) {
+	return (X[0] - 1) * (X[0] - 2) * (X[0] - 3);
+}
+
+NT logx_natural_1D(Point X) {
+	return log(X[0]);
+}
+
+NT rooted_squaresum(Point X) {
+	return sqrt(X.squared_length());
+}
+
+NT one_sqsum(Point X) {
+	return 1 - X.squared_length();
+}
+
+template <typename NT>
+void test_values (NT computed, NT expected, NT exact) {
+	std::cout << "Computed integration value = " << computed << std::endl;
+	std::cout << "Expected integration value = " << expected << std::endl;
+	std::cout << "Exact integration value = " << exact << std::endl;
+	std::cout << "Relative error (expected) = " << std::abs((computed - expected)/expected) << std::endl;
+	std::cout << "Relative error (exact) = " << std::abs((computed - exact)/exact) << std::endl ;
+    CHECK(((std::abs((computed - expected)/expected) < 0.00001) || (std::abs((computed - exact)/exact) < 0.2)));
+}
+
+template <typename NT>
+void call_test_simple_mc_integration_over_rectangles() {
```

Comment:
this contains integration over segments too, so this could be included in the name